api.planx.uk/modules/analytics/metabase/dashboard/updateFilter.ts

```diff
@@ -0,0 +1,52 @@
+import { $metabase } from "../shared/client.js";
+import type {
+  UpdateFilterParams,
+  UpdateFilterResponse,
+  FilterParam,
+} from "./types.js";
+
+/** Takes the ID of the dashboard to be updated, the name of the filter (a string, must be an exact match), and the new value to be filtered on.
+ * Currently only works for strings. */
+export async function updateFilter(
+  params: UpdateFilterParams,
+): Promise<UpdateFilterResponse> {
+  // Get existing dashboard data
+  const response = await $metabase.get(`/api/dashboard/${params.dashboardId}`);
+
+  // Update filter default value parameter
+  let updatedFilter;
+  const updatedParameters = response.data.parameters.map(
+    (param: FilterParam) => {
```

Comment:
We need the type here as `response` is untyped. If we add the response type to the `$metabase.get<T>()` function, this shouldn't be needed here.